### PR TITLE
Properly handle multiline comments (or any other tokens that span more than one line)

### PR DESCRIPTION
**Pull request by k3rni**
This adresses #3044, and possibly also #2891, although I haven't tested for that yet.

The solution is to check if a returned token has newlines inside, and if so, produce as many style span elements as there are lines, following each one with a newline. This in turn gets properly wrapped in line-numbers.

### CODE REVIEW

**Comment by kytrinyx** [comment]
I get the correct syntax highlighting, but I also get a lot of weird/extra whitespace at the top and the bottom of the submission.

screen shot 2016-08-11 at 10 39 08 am

**Comment by kytrinyx** [comment]
The python example was fixed for the multi-line comment, but also had whitespace issues.

Submitted:

```
"""
Implementation of the "word-count" exercise.
"""


def word_count(phrase):
    book = {}
    for word in phrase.split():
        if word not in book:
            book[word] = 0
        book[word] += 1
    return book
```

Notice that there is no blank line between the comment and the function definition:

screen shot 2016-08-11 at 10 45 03 am

File: test/exercism/html_formatter_test.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require_relative '../test_helper'
+require 'rouge'
+require 'loofah'
+require_relative '../../lib/rouge/formatters/html_exercism'
+
+class HTMLFormatterTest < MiniTest::Test
+  def setup
```
Comment (bernardoamc) [inline test/exercism/html_formatter_test.rb]: Can we ✂️ this empty `setup`?

**Comment by dcorking** [comment]
At first glance, it doesn't seem wise that rouge emits one `span` element for multiple lines of `pre`formatted code. Any idea how interested the rouge project might be in a pull request to change that to one `span` per line? It might make exercism's code cleaner.

**Comment by k3rni** [comment]
Thanks for your feedback! The problems were not with rouge itself, it only emits tokens. It's the formatters' job to wrap them properly in newlines.

**Comment by Insti** [comment]
Rouge since 2.0 does have several different formatters, it may be that one of them is more appropriate than the one we are currently using.

**Comment by k3rni** [comment]
The failures left at this point are all about leading spaces being highlighted with `<span class="w">` in *some* languages (notably Clojure) and other whitespace issues.

**Comment by kytrinyx** [comment]
> Is it okay to adjust the tests to accept this whitespace change?

Yeah, I think that's fine.

I'll review this over the weekend. Thanks!

**Comment by kytrinyx** [comment]
Merged in ea4cafbbe4bda84928e08f1e69091f768ff28e14

Thank you so much for taking the time to solve this so thoroughly. :heart:

**Pull request closed without merging**